app/main.py

```diff
@@ -1,2 +1,89 @@
 class Dictionary:
-    pass
+    def __init__(self, size=8):
+        self.storage = [[] for _ in range(size)]
+        self.size = size
+        self.length = 0
+        self.load_factor = 2 / 3
+
+    def __setitem__(self, key, value):
+        storage_idx = hash(key) % self.size
+
+        while True:
+            if not self.storage[storage_idx]:
+                self.length += 1
+                self.storage[storage_idx] = [key, value, hash(key)]
+                break
+            elif self.storage[storage_idx][0] == key:
+                self.storage[storage_idx][1] = value
+                break
+            storage_idx = (storage_idx + 1) % self.size
+
+        if self.length >= self.load_factor * self.size:
+            self.resize(self.storage)
```

Comment:
this logic should be at the start of the function

Reply:
If I move this to the top I have a problem with resize method. I tried but then should be different implementation